cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -161,26 +159,49 @@ private void downloadFileWithProgressBar(String url, Path target, HttpResponse<I
    *
    * @param source Path of file to copy
    * @param target Path of target directory
+   * @param test the flag about calling this method from test classes.
    */
-  private void copyFileWithProgressBar(Path source, Path target) throws IOException {
+  private void copyFileWithProgressBar(Path source, Path target, boolean test) throws IOException {
 
     try (InputStream in = new FileInputStream(source.toFile()); OutputStream out = new FileOutputStream(target.toFile())) {
+      long size;
+      if (test) {
+        size = 0L;
+      } else {
+        size = source.toFile().length();
+      }
 
-      long size = source.toFile().length();
+      informAboutSettingDefaultContentLength(size, null, source);
       byte[] buf = new byte[1024];
       int readBytes;
 
       try (IdeProgressBar pb = this.context.prepareProgressBar("Copying", size)) {
         while ((readBytes = in.read(buf)) > 0) {
           out.write(buf, 0, readBytes);
-          pb.stepByOne();
+          if (size > 0) {
+            pb.stepBy(readBytes);
+          }
         }
       } catch (Exception e) {
         throw new RuntimeException(e);
       }
     }
   }
 
+  private void informAboutSettingDefaultContentLength(long contentLength, String url, Path path) {
+
+    String source;
+    if (contentLength == 0) {
```

Comment:
why are you using `0` for undefined?
Files can be `0` bytes large and then the user should not get this warning for every file that was empty but gets copied.
Also if the file size is know but smaller than e.g. 1k we should not give a progress bar.
What is good is that we have a dedicated method `copyFileWithProgressBar` for this use-case so we do not do this for every file. However, I assume this is used when we get a tool release package from a custom tool repository using a file URL and that can also be a network mount that is slower so already copying 1MB could take some seconds...

Reply:
I've set the default for undefined to -1 now.